sdk/kiosk/src/tx/kiosk.ts

```diff
@@ -0,0 +1,310 @@
+ *
+ * Unlike `place` this function requires a `TransferPolicy` to exist
+ * and be passed in. This is done to make sure the item does not get
+ * locked without an option to take it out.
+ */
+export function lock(
+  tx: TransactionBlock,
+  itemType: string,
+  kiosk: ObjectArgument,
+  kioskCap: ObjectArgument,
+  policy: ObjectArgument,
+  item: ObjectArgument,
+) {
+  tx.moveCall({
+    target: `${KIOSK_MODULE}::lock`,
+    typeArguments: [itemType],
+    arguments: [
+      objArg(tx, kiosk),
+      objArg(tx, kioskCap),
+      objArg(tx, policy),
+      objArg(tx, item),
+    ],
+  });
+}
+
+/**
+ * Call the `kiosk::take<T>(Kiosk, KioskOwnerCap, ID)` function.
+ * Take an item from the Kiosk.
+ */
+export function take(
+  tx: TransactionBlock,
+  itemType: string,
+  kiosk: ObjectArgument,
+  kioskCap: ObjectArgument,
+  itemId: SuiAddress,
+): TransactionArgument {
+  let [item] = tx.moveCall({
+    target: `${KIOSK_MODULE}::take`,
+    typeArguments: [itemType],
+    arguments: [
+      objArg(tx, kiosk),
+      objArg(tx, kioskCap),
+      tx.pure(itemId, 'address'),
+    ],
+  });
+
+  return item;
+}
+
+/**
+ * Call the `kiosk::list<T>(Kiosk, KioskOwnerCap, ID, u64)` function.
+ * List an item for sale.
+ */
+export function list(
+  tx: TransactionBlock,
+  itemType: string,
+  kiosk: ObjectArgument,
+  kioskCap: ObjectArgument,
+  itemId: SuiAddress,
+  price: string | bigint,
+) {
+  tx.moveCall({
+    target: `${KIOSK_MODULE}::list`,
+    typeArguments: [itemType],
+    arguments: [
+      objArg(tx, kiosk),
+      objArg(tx, kioskCap),
+      tx.pure(itemId, 'address'),
+      tx.pure(price, 'u64'),
+    ],
+  });
+
+  return tx;
+}
+
+/**
+ * Call the `kiosk::place_and_list<T>(Kiosk, KioskOwnerCap, Item, u64)` function.
+ * Place an item to the Kiosk and list it for sale.
+ */
+export function placeAndList(
+  tx: TransactionBlock,
+  itemType: string,
+  kiosk: ObjectArgument,
+  kioskCap: ObjectArgument,
+  item: ObjectArgument,
+  price: string | bigint,
+) {
+  tx.moveCall({
+    target: `${KIOSK_MODULE}::place_and_list`,
+    typeArguments: [itemType],
+    arguments: [
+      objArg(tx, kiosk),
+      objArg(tx, kioskCap),
+      objArg(tx, item),
+      tx.pure(price, 'u64'),
+    ],
+  });
+
+  return tx;
+}
+
+/**
+ * Call the `kiosk::purchase<T>(Kiosk, ID, Coin<SUI>)` function and receive an Item and
+ * a TransferRequest which needs to be dealt with (via a matching TransferPolicy).
+ */
+export function purchase(
+  tx: TransactionBlock,
+  itemType: string,
+  kiosk: ObjectArgument,
+  itemId: SuiAddress,
+  payment: ObjectArgument,
+): [TransactionArgument, TransferRequest] {
+  let [item, transferRequest] = tx.moveCall({
+    target: `${KIOSK_MODULE}::purchase`,
+    typeArguments: [itemType],
+    arguments: [
+      objArg(tx, kiosk),
+      tx.pure(itemId, 'address'),
+      objArg(tx, payment),
+    ],
+  });
+
+  return [item, { ...transferRequest, itemType } as TransferRequest];
```

Comment:
You should really treat the return values out of `moveCall` as opaque values, don't spread it here and instead just have two properties that folks need to use